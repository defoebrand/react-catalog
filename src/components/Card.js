// import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import '../styles/Card.scss';

const Card = ({ entry }) => {
  const handleClick = () => {
    console.log(entry);
  };

  return (
    <>
      <button type="button" key={entry.id} className="Card" onClick={() => handleClick()}>
        {/* console.log(entry) */}
        <p>{entry.name}</p>
        <img src={entry.images.lg} alt={entry.name} />
      </button>
    </>

  );
};

Card.propTypes = {
  entry: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    images: PropTypes.shape({
      lg: PropTypes.string,
    }),
  }),
};

Card.defaultProps = {
  entry: {},

};

// export default connect(state => ({ books: state.books }))(Card);

export default Card;
