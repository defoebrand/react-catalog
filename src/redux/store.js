import { combineReducers, createStore } from 'redux';

import { SUBMIT, CREATE_ENTRY, DISPLAY_ENTRY } from './actions';

const initialState = '';

const initialHeroes = [];

const superHeroReducer = (state = initialHeroes, action) => {
  switch (action.type) {
    case CREATE_ENTRY:
      // console.log('create entry state', state);
      // console.log('create entry action', action);
      // localStorage.entries = JSON.stringify(action.entries);
      return {
        entries: action.entries,
      };

    case DISPLAY_ENTRY:
      console.log('display entry state', state);
      console.log('display entry action', action);
      // localStorage.entries = JSON.stringify(action.entry);
      return {
        entry: action.entry,
        display: 'singleCard',
      };

    default:
      // console.log('default entry state', state);
      return state;
  }
};

const searchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SUBMIT:
      // console.log('submit state', state.filter);
      // console.log('submit action', action);
      return {
        filter: action.input,
        display: 'manyCards',
      };

    default:
      // console.log('default submit state', state.filter);
      return state;
  }
};

const combinedReducers = combineReducers({
  superHeroReducer,
  searchReducer,
});

export default createStore(combinedReducers);
