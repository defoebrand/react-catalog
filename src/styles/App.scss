.App {
  text-align: center;
  // width: 100%;
  // display: flex;
  // flex-wrap: wrap;
}
