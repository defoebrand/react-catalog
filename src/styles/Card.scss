.Card {
  border: 1px solid white;
  margin: 10px;
  padding: 10px;
  cursor: pointer;
  // overflow: hidden;

  iframe {
    // height: 300px;
    // width: 150px;
    // margin-top: -150px;
  }

  img {
    max-width: 100%;
  }
}
