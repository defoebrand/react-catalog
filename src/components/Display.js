import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useEffect } from 'react';

import {
  displayEntry, displayEntries,
} from '../redux/actions';

import Card from './Card';

import '../styles/Display.scss';

const Display = ({
  filter, entries, dispatch, history,
}) => {
  useEffect(() => {
    dispatch(displayEntries('manyCards'));
  }, []);

  const handleClick = character => {
    dispatch(displayEntry(character));
    history.push(`/stats/${character.name}`);
  };

  const regex = new RegExp(filter, 'i');

  const dataMap = (
    filter === ''
      ? (entries.map(entry => (
        <Card key={entry.slug} entry={entry} clickEvent={handleClick} />)))
      : (entries.filter(entry => (regex.test(entry.name))).map(entry => (
        <Card key={entry.slug} entry={entry} clickEvent={handleClick} />)))
  );

  return (
    <main className="Display">
      {dataMap}
    </main>
  );
};

Display.propTypes = {
  filter: PropTypes.string,
  entries: PropTypes.arrayOf(PropTypes.shape()),
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.shape().isRequired,
};

Display.defaultProps = {
  filter: '',
  entries: [],
};

export default connect(state => ({
  filter: state.searchReducer.filter,
  entries: state.superHeroReducer.entries,
}))(Display);
