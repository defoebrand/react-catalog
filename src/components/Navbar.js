import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { runSearch, createEntries } from '../redux/actions';

import '../styles/Navbar.scss';

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
    this.handleChange.bind(this);
    this.handleClick.bind(this);

    // fetch('https://akabab.github.io/superhero-api/api/all.json').then(response => response.json()).then(data => {
    //   console.log(data);
    //   // dispatch(createEntries(data));
    // }).catch(err => err);
  }

  componentDidMount() {
    const { dispatch } = this.props;
    fetch('https://akabab.github.io/superhero-api/api/all.json').then(response => response.json()).then(data => {
      console.log(data);

      localStorage.entries = JSON.stringify(data);

      dispatch(createEntries(data));
      // console.log(supers);
    }).catch(err => err);
  //   document.body.addEventListener('keypress', event => {
  //     const { input } = this.state;
  //     if (event.key === 'Enter') {
  //       dispatch(runSearch(input));
  //       this.setState({
  //         input: '',
  //       });
  //     }
  //   });
  }

    handleChange = event => {
      const { dispatch } = this.props;

      this.setState({
        input: event.target.value,
      });
      dispatch(runSearch(event.target.value));
    }

    handleClick = () => {
      const { input } = this.state;
      const { dispatch } = this.props;
      dispatch(runSearch(input));
      this.setState({
        input: '',
      });
    }

    render() {
      const { input } = this.state;

      return (
        <header className="Navbar">
          <h1>Hello!</h1>
          <input value={input} onChange={this.handleChange} />
          <input type="submit" value="Submit" onClick={this.handleClick} />
        </header>
      );
    }
}
Navbar.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect(null)(Navbar);
