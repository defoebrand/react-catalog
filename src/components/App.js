import '../styles/App.scss';
import Navbar from './Navbar';

const App = () => (
  <div className="App">
    <Navbar />
    <p>
      Hello World!
    </p>
  </div>
);

export default App;
