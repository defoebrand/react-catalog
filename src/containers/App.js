import { Route } from 'react-router-dom';

import Navbar from '../components/Navbar';
import Display from '../components/Display';
import Card from '../components/Card';

import '../styles/App.scss';

const App = () => (
  <div className="App">
    <Route path="/" component={Navbar} />
    <Route path="/all" component={Display} />
    <Route path="/stats/:super" component={Card} />
  </div>
);

export default App;
