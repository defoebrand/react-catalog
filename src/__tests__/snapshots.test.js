import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';

import App from '../containers/App';
import Navbar from '../components/Navbar';
import Card from '../components/Card';
import Display from '../components/Display';

describe('spanshots', () => {
  it('renders App correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const history = {
    location: {
      pathname: '/all',
    },
  };

  it('renders Navbar correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Navbar history={history} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const entry = {
    slug: '1-SuperHero',
    name: 'Superhero',
    biography: {
      alignment: 'good',
    },
    images: {
      sm: '../assets/batLogo.png',
    },
  };

  it('renders Card correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Card key={entry.slug} entry={entry} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Card correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Display filter="" entries={entry} history={history} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
