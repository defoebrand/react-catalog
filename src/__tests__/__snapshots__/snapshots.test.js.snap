// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spanshots renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="Navbar"
  >
    <button
      className="batLogo"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="BatComputer"
        src="batLogo.png"
      />
    </button>
    <div>
      <h1>
        BatComputer
      </h1>
      <input
        onChange={[Function]}
      />
    </div>
    <button
      className="batLogo"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="BatComputer"
        src="batLogo.png"
      />
    </button>
  </header>
</div>
`;

exports[`spanshots renders Card correctly 1`] = `
<button
  className="manyCards"
  onClick={[Function]}
  style={
    Object {
      "border": "3px solid blue",
    }
  }
  type="button"
>
  <p>
    Superhero
  </p>
  <img
    alt="Superhero"
    src="../assets/batLogo.png"
  />
</button>
`;

exports[`spanshots renders Card correctly 2`] = `
<main
  className="Display"
/>
`;

exports[`spanshots renders Navbar correctly 1`] = `
<header
  className="Navbar"
>
  <button
    className="batLogo"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="BatComputer"
      src="batLogo.png"
    />
  </button>
  <div>
    <h1>
      BatComputer
    </h1>
    <input
      onChange={[Function]}
    />
  </div>
  <button
    className="batLogo"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="BatComputer"
      src="batLogo.png"
    />
  </button>
</header>
`;
